tools/kagami/src/main.rs

```diff
@@ -849,13 +853,69 @@ mod validator {
         }
     }
 
+    pub fn clone_default_validator(out_dir: impl AsRef<Path>) -> Outcome {
+        const GIT_ORIGIN: &str = "https://github.com/hyperledger/iroha.git";
+
+        // Sparse directory copy as suggested in https://stackoverflow.com/a/52269934/10669935
+        cmd!(
+            "git",
+            "clone",
+            "-n",
+            "--depth=1",
+            "--filter=tree:0",
+            "--branch=iroha2-dev",
+            GIT_ORIGIN,
+            out_dir.as_ref()
+        )
+        .dir(out_dir.as_ref())
+        .run()?;
+        // `--no-cone` to make it an include list
+        cmd!(
+            "git",
+            "sparse-checkout",
+            "set",
+            "--no-cone",
+            "/Cargo.toml",
+            "crypto",
+            "data_model",
+            "default_validator",
+            "derive_primitives",
+            "ffi",
+            "macro",
+            "primitives",
+            "schema",
+            "version",
+            "wasm",
+        )
+        .dir(out_dir.as_ref())
+        .run()?;
```

Comment:
Not really happy here with how many packages from workspace it required to pull up, which is mostly due to `wasm` crate dependencies. Maybe it would make sense to introduce a new feature in that crate so that it would allow compiling only `wasm/validator`, which would make the pull slimmer. For now I've chosen the route of minimal change.

Reply:
Do we need to optimise it this way?

This partial clone is really hard to cover with tests, thus it is very error prone: the workspace configuration might change and it might be far from obvious that this part will break.

I think it will be simpler and less error-prone if we work with a whole repo.

Reply:
I can see this later on, when we have stabilised the libraries. But right now, I'd rather you did what @0x009922 asks you to do.